### DISCUSSION TITLE
[File Service] Introduce API to move files across buckets (similar to renaming)

### DISCUSSION THREAD

**Discussion by danibix95** [mia-platform/community#202] [Ideas]
File Service plugin is going soon to be able to manage multiple buckets or different folders within the same bucket. An useful feature would be the possibility to move files across buckets using a single API request.

A possible API request, adapting a similar format to the one [already available](https://docs.mia-platform.eu/docs/runtime_suite/files-service/usage), could be the following:

```
PATCH http://file-service/:file
{
  source: <bucketOrigin>
  destination: <bucketDestination>
}
```

**Comment by fredmaggiowski** [comment]
Hi. interesting feature, I wonder why it is needed though, is it for migration purposes? Why the file hasn't be uploaded in the correct bucket in the first place? 🤔

**Reply by danibix95** [reply]
I am planning to use the multiple bucket feature of File Service to manage two different folders within the same bucket. Adding this feature would allow my service to move the file from one folder to another without including further bucket logic in it.

Eventually, I would be happy also with just the possibility to manage and move files across different folders of the same bucket

**Reply by fredmaggiowski** [reply]
folders management is not in the roadmap for the files service, scopes managed with multiple buckets may be used to achieve something like that on your crud metadata but not between the same bucket. Multibucket support may help you loading different files in different buckets but always with a flat approach
